src/protocols/typed-message/debugger.tsx

```diff
@@ -0,0 +1,87 @@
+// @ts-ignore
+const React = require('jsx-jsonml-devtools-renderer') as any
+import type { TypedMessage, TypedMessageCompound, TypedMessageImage, TypedMessageText } from './types'
+
+class TypedMessageFormatter {
+    isTypedMessage(obj: unknown): obj is TypedMessage {
+        if (typeof obj !== 'object' || obj === null) return false
+        if (!('version' in obj)) return false
+        if (!('meta' in obj)) return false
+        if (!('type' in obj)) return false
+        return true
+    }
+    hasBody(obj: unknown) {
+        if (!this.isTypedMessage(obj)) return false
+        if (obj.type === 'empty') return false
+        return true
+    }
+    compound(obj: TypedMessageCompound) {
+        return (
+            <div style={{ maxWidth: '95vw', overflow: 'break-word' }}>
+                <ol>
+                    {obj.items.map((x) => (
+                        <li>{display(x)}</li>
```

Comment:
`key` is lost.

Reply:
This file is not React, it's a custom JSX framework. the render target does not support "key".

This file renders TypedMessage in the Chrome dev tools.

![](https://user-images.githubusercontent.com/5390719/90721983-8503cf00-e2ec-11ea-9cb8-ac89db84dc2d.png)